src/GraphQL.Tests/Bugs/Bug1699InvalidEnum.cs

```diff
@@ -0,0 +1,146 @@
+
+        [Fact]
+        public void Input_Enum_Valid() => AssertQuerySuccess("{ inputEnum(arg: SLEEPY) }", @"{ ""inputEnum"": ""SLEEPY"" }");
+
+        [Fact]
+        public void Input_Enum_Valid2() => AssertQuerySuccess("{ input(arg: SLEEPY) }", @"{ ""input"": ""Sleepy"" }");
+
+        [Fact]
+        public void Input_Enum_InvalidEnum() => AssertQueryWithError("{ input(arg: DOPEY) }", null, "Argument \u0022arg\u0022 has invalid value DOPEY.\nExpected type \u0022Bug1699Enum\u0022, found DOPEY.", 1, 9, (object[])null, code: "5.3.3.1");
+
+        [Fact]
+        public void Input_Enum_InvalidString() => AssertQueryWithError(@"{ input(arg: ""SLEEPY"") }", null, "Argument \u0022arg\u0022 has invalid value \u0022SLEEPY\u0022.\nExpected type \u0022Bug1699Enum\u0022, found \u0022SLEEPY\u0022.", 1, 9, (object[])null, code: "5.3.3.1");
+
+        [Fact]
+        public void Input_Enum_InvalidInt() => AssertQueryWithError(@"{ input(arg: 2) }", null, "Argument \u0022arg\u0022 has invalid value 2.\nExpected type \u0022Bug1699Enum\u0022, found 2.", 1, 9, (object[])null, code: "5.3.3.1");
+
+        [Fact]
+        public void Input_Enum_Valid_Variable() => AssertQuerySuccess("query($arg: Bug1699Enum!) { input(arg: $arg) }", @"{ ""input"": ""Grumpy"" }", "{\"arg\":\"GRUMPY\"}".ToInputs());
+
+        [Fact]
+        public void Input_Enum_InvalidEnum_Variable() => AssertQueryWithError(@"query($arg: Bug1699Enum!) { input(arg: $arg) }", null, "Variable \u0027$arg\u0027 is invalid. Unable to convert \u0027DOPEY\u0027 to Bug1699Enum", 0, 0, (object[])null, code: "INVALID_VALUE", inputs: "{\"arg\":\"DOPEY\"}");
+
+        [Fact]
+        public void Input_Enum_InvalidInt_Variable() => AssertQueryWithError(@"query($arg: Bug1699Enum!) { input(arg: $arg) }", null, "Variable \u0027$arg\u0027 is invalid. Unable to convert \u00272\u0027 to Bug1699Enum", 0, 0, (object[])null, code: "INVALID_VALUE", inputs: "{\"arg\":2}");
+
+        [Fact]
+        public void Input_Enum_UndefinedDefault() => AssertQuerySuccess("{ input }", @"{ ""input"": ""Grumpy"" }");
+
+        [Fact]
+        public void Input_Enum_SetDefault() => AssertQuerySuccess("{ inputWithDefault }", @"{ ""inputWithDefault"": ""Happy"" }");
+
+        [Fact]
+        public void Input_Enum_OverrideDefault() => AssertQuerySuccess("{ inputWithDefault(arg: SLEEPY) }", @"{ ""inputWithDefault"": ""Sleepy"" }");
+
+        [Fact]
+        public void Input_Enum_MissingRequired() => AssertQueryWithError(@"{ inputRequired }", null, "Field \u0022inputRequired\u0022 argument \u0022arg\u0022 of type \u0022Bug1699Enum!\u0022 is required but not provided.", 1, 3, (object[])null, code: "5.3.3.2");
+
+        [Fact]
+        public void Input_Enum_RequiredWithDefault() => AssertQueryWithError(@"{ inputRequiredWithDefault }", null, "Field \u0022inputRequiredWithDefault\u0022 argument \u0022arg\u0022 of type \u0022Bug1699Enum!\u0022 is required but not provided.", 1, 3, (object[])null, code: "5.3.3.2");
```

Comment:
It seems wrong to me. See http://spec.graphql.org/June2018/#sec-Required-Arguments . The numbering of the validation rules is a little off, but it doesn't matter.

> Arguments can be required. An argument is required if the argument type is non‐null **and does not have a default value**. Otherwise, the argument is optional.

Reply:
Makes sense.  I'll need to review the code to see where to implement the fix.

Reply:
Done -- fixed validation rule for this.  See commit below.

Reply:
It seems that I fixed it too in one of my draft PR ). Otherwise I would not pay attention to this part. Will check my commits later to validate changes. Probably we will close one more issue.

Reply:
No, I fixed another rule. The problem is similar, the default value for the argument is ignored.